src/server/game/Spells/Auras/SpellAuraEffects.cpp

```diff
@@ -528,9 +528,14 @@ int32 AuraEffect::CalculateAmount(Unit* caster)
             {
                 Item* item1 = caster->ToPlayer()->GetWeaponForAttack(BASE_ATTACK);
                 Item* item2 = caster->ToPlayer()->GetWeaponForAttack(OFF_ATTACK);
```

Comment:
I think you can just add next lines and everything else leave untouched:
```
if (!item2)
    item2 = caster->ToPlayer()->GetShield();
```

Reply:
Or even better:
```
Item* item2 = caster->ToPlayer()->GetItemByPos(INVENTORY_SLOT_BAG_0, EQUIPMENT_SLOT_OFFHAND);
```
But I'm not sure about some possible edge cases.

Reply:
> I think you can just add next lines and everything else leave untouched:
> 
> ```
> if (!item2)
>     item2 = caster->ToPlayer()->GetShield();
> ```

This worked, your other suggestion didnt seem to. So I've changed it to this null check

Reply:
@ChrisCGalbraith Have you tried this one?
```
Item* item1 = caster->ToPlayer()->GetWeaponForAttack(BASE_ATTACK);
Item* item2 = caster->ToPlayer()->GetWeaponForAttack(OFF_ATTACK);
if (!item2)
    item2 = caster->ToPlayer()->GetShield();

if (item1 && item2 && (item1->GetTemplate()->InventoryType == INVTYPE_2HWEAPON || item2->GetTemplate()->InventoryType == INVTYPE_2HWEAPON))
    amount = -10;
else
    amount = 0;
```

Reply:
Removed the unnecessary OR, think this should be it now